lib/utils/types_utils.dart

```diff
@@ -35,6 +36,12 @@ bool hasWidgetType(DartType type) =>
         _isListInheritedProvider(type) ||
         _isFutureInheritedProvider(type));
 
+bool isIterableOrSubclass(DartType? type) =>
+    _checkSelfOrSupertypes(type, (t) => t?.isDartCoreIterable ?? false);
```

Comment:
All subclasses of Iterable are also Iterable.
```suggestion
bool isIterable(DartType? type) =>
    _checkSelfOrSupertypes(type, (t) => t?.isDartCoreIterable ?? false);
```